zebra-rpc/src/methods.rs

```diff
@@ -601,6 +728,30 @@ pub struct GetBlock(#[serde(with = "hex")] SerializedBlock);
 #[derive(Copy, Clone, Debug, Eq, PartialEq, serde::Deserialize, serde::Serialize)]
 pub struct GetBestBlockHash(#[serde(with = "hex")] block::Hash);
 
+/// Response to a `z_gettreestate` RPC request.
+///
+/// Contains the hex-encoded Sapling & Orchard note commitment trees.
+// TODO: adjust the description above so that it reflects the new fields once
+// they are added.
+#[derive(Clone, Debug, Eq, PartialEq, serde::Deserialize, serde::Serialize)]
+pub struct GetTreestate {
+    hash: block::Hash,
+    height: block::Height,
+    time: DateTime<Utc>,
+    sapling_tree: NoteCommitmentTree,
+    orchard_tree: NoteCommitmentTree,
```

Comment:
Minor: Since the `NoteCommitmentTree` enum allows both Sapling and Orchard, we could accidentally supply two Sapling trees here.

```suggestion
    sapling_tree: Option<sapling::tree::NoteCommitmentTree>,
    orchard_tree: Option<orchard::tree::NoteCommitmentTree>,
```

Reply:
I didn't realize this would be an issue. I wrapped both trees into one enum because I was thinking that would make the subsequent serialization easier.

Reply:
Fixed in [37aa838](https://github.com/ZcashFoundation/zebra/pull/3990/commits/37aa8385269794952b8fe00357e723d93e085041).

Reply:
> I wrapped both trees into one enum because I was thinking that would make the subsequent serialization easier.

Let me know if you need help with the exact serialization structure, it's a bit tricky the first time around!